openmc/mesh.py

```diff
@@ -1544,12 +1546,12 @@ class SphericalMesh(StructuredMesh):
 
     def __init__(
         self,
-        mesh_id: typing.Optional[int] = None,
+        origin: Tuple[float] = (0., 0., 0.),
+        r_grid: Sequence[float] = (0, 10),
```

Comment:
Same thing re: grids and origin

Reply:
Thanks Paul, how about z_grid?

Reply:
Right, z_grid also doesn't have a sensible default in my opinion, so we should make that required as well